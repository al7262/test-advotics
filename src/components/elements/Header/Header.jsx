import React from 'react';
import styles from './index.css';
import logo from '../../../assets/logo-full.jpg';
import profile from '../../../assets/profile.svg';
import logout from '../../../assets/logout.png';

export default function Header() {
  return (
    <header className={styles.root}>
      <figure>
        <img src={logo} alt="advotics" />
      </figure>
      <figcaption>
        <span>powered by</span>
        <img src={logo} alt="advotics" />
      </figcaption>
      <div className={styles.profile}>
        <h6>Username</h6>
        <small>Company Name</small>
        <img src={profile} alt="profile-company" />
        <img src={logout} alt="logout-advotics" />
      </div>
    </header>
  );
}
