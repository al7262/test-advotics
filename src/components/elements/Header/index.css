.root {
    position: fixed;
    width: 100vw;
    height: 4em;
    padding: 1em 2em;
    background-color: var(--white-color);
    display: flex;
    gap: .5em;
    box-shadow: 0px .125em .375em var(--black-color-3);
    z-index: 10;
}

.root > figure {
    height: 100%;
}

.root > figure > img {
    height: 100%;
}

.root > figcaption {
    display: flex;
    align-items: flex-end;
    justify-content: center;
}

.root > figcaption > img {
    height: 1.125em;
    margin-left: .25em;
}

.profile {
    display: grid;
    margin-left: auto;
    grid-template-rows: auto auto;
    column-gap: 1.25em;
    align-items: center;
    text-align: right;
}

.profile > h6 {
    grid-row: 1;
}

.profile > small {
    color: var(--black-color-3)
}

.profile > img {
    grid-row: 1 / span 2;
}

.profile > img:first-of-type {
    width: 2em;
    height: 2em;
}

.profile > img:nth-of-type(2) {
    width: 1em;
    height: 1em;
    margin-left: .125em;
}