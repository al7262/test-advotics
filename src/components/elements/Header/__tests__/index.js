/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../Header';

describe('src/components/elements/Header', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Header />);
    expect(tree).toMatchSnapshot();
  });
});
