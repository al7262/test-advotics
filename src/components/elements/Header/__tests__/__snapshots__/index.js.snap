// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Header render 1`] = `
<header>
  <figure>
    <img
      alt="advotics"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </figure>
  <figcaption>
    <span>
      powered by
    </span>
    <img
      alt="advotics"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </figcaption>
  <div>
    <h6>
      Username
    </h6>
    <small>
      Company Name
    </small>
    <img
      alt="profile-company"
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      alt="logout-advotics"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</header>
`;
